actors/market/src/lib.rs

```diff
@@ -879,7 +879,7 @@ fn alloc_request_for_deal(
     let alloc_expiration =
         min(deal.proposal.start_epoch, curr_epoch + policy.maximum_verified_allocation_expiration);
     ext::verifreg::AllocationRequest {
-        provider: deal.proposal.provider,
+        provider: deal.proposal.provider.id().unwrap(),
```

Comment:
This is ok, but please add a brief comment to this method noting that the proposal must have ID addresses.